src/Symfony/Component/Yaml/Inline.php

```diff
@@ -156,12 +156,22 @@ public static function dump($value, $exceptionOnInvalidType = false, $objectSupp
      */
     private static function dumpArray($value, $exceptionOnInvalidType, $objectSupport)
     {
+        if ($value) {
```

Comment:
I was intrigued by this problem: *"how to detect if an array is associative or not"* and I found two one-liners on StackOverflow. Maybe we can simplify this code?

```php
$isMapping = array_keys($value) !== range(0, count($value) - 1);

$isMapping = count(array_filter(array_keys($value), 'is_string')) > 0;
```